### PR TITLE
chore: check typescript error on push

### PR DESCRIPTION
**Pull request by cguedes**
Add a new yarn command to check typescript errors.

```bash
yarn ts:check
```

This command/check was also added to the git push hook to ensure that we can't push code that don't compile.

Also updates the `yarn check` to include this check.

Related to #24 and #10 

@sehyod

### CODE REVIEW

File: .husky/pre-push [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 yarn lint:check
```
Comment (danvk) [inline .husky/pre-push:4]: Can this just run `yarn check`? What's the distinction here?

Comment (cguedes) [reply .husky/pre-push:4]: I was wondering exactly that.
In a way we have more control (and legibility) over what get executed on push, but we can also forget to add them here!

So I'm ok to update this to use `yarn check` instead.

What do you think @sehyod ?

Comment (sehyod) [reply .husky/pre-push:4]: I agree with using `yarn check` in the pre-push hook: it's better not to have to update the code in multiple places every time we want to add a new check

**Review by danvk — APPROVED** [review]

**Comment by cguedes** [comment]
@sergioramos this is ready to be merged.

**Pull request merged**